packages/oauth/oauth-provider/src/router/create-api-middleware.ts

```diff
@@ -451,20 +471,21 @@ export function createApiMiddleware<
             onError?.(req, res, err, 'Failed to delete request')
           }
 
-          if (err instanceof AccessDeniedError && err.parameters.redirect_uri) {
-            // Prefer logging the cause
-            onError?.(req, res, err.cause ?? err, 'Authorization failed')
-
-            const url = buildRedirectUrl(
-              server.issuer,
-              err.parameters,
-              err.toJSON(),
-            )
+          if (err instanceof AccessDeniedError) {
+            try {
+              const url = buildRedirectUrl(
+                server.issuer,
+                err.parameters,
+                err.toJSON(),
+              )
 
-            return { url }
+              return { json: { url } }
+            } catch {
+              // Unable to build redirect URL, ignore
+            }
           }
 
-          throw err
+          return buildErrorJsonResponse(err)
```

Comment:
Because we already logged the error (`onError` above), we don't want to re-throw here (to avoid double logging of the error)